src/Internal/Calculator/NativeCalculator.php

```diff
@@ -210,6 +210,34 @@ public function pow(string $a, int $e) : string
         return $result;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function sqrt(string $a) : string
+    {
+        $x = $a;
+        if ($a[0] === '-') {
+            return NAN;
```

Comment:
NAN is a `float`, not a string; you cannot return NAN. We should throw an exception here. Actually we can even *do nothing*: `Calculator` is not part of the public API, and assumes that all input is already validated.

As such, we can perform the test only in the `BigNumber` classes, you can safely remove this check.